File: Java/src/model/Expresions/ReadExpression.java
package model.Expresions;

import Exceptions.ValueNotFoundException;
import interfaces.Expression;
import interfaces.IDictionary;
import interfaces.IStatement;

import java.util.Scanner;

/**
 * Created by Lucian on 11/9/2015.
 */
public class ReadExpression implements Expression {

    /**
     * The number
     */
    private int number;


    /**
     * The constructor
     *
     */
    public ReadExpression() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("\nENTER YOUR NUMBER: ");
        number = scanner.nextInt();
    }


    @Override
    public int eval(IDictionary<String, Integer> table) throws ValueNotFoundException {
        return number;
    }

    /**
     * String representation
     *
     * @return String
     */
    @Override
    public String toString() {
        return "Read(" + String.valueOf(number) + ")";

    }
}

File: C#/ToyLanguage/ToyLanguage/Model/Expressions/ReadExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguage.Interfaces;

namespace ToyLanguage.Model.Expressions
{
    [Serializable]
    class ReadExpression : IExpressions
    {

        /**
         * The number
         */
        private int number;


        /**
         * The constructor
         *
         */
        public ReadExpression() {
 
            Console.WriteLine("\nENTER YOUR NUMBER: ");
            number = Int32.Parse(Console.ReadLine());
        }

        public int eval(IMyDictionary<string, int> table)
        {
            return number;
        }

      
        /**
         * String representation
         *
         * @return String
         */
     public String MyToString() {
            return "Read(" + number.ToString() + ")";

        }
    }
}
